Gems/ONNX/Code/Source/Clients/Model.cpp

```diff
@@ -12,10 +12,27 @@ namespace ONNX
 {
     void Model::Load(const InitSettings& initSettings)
     {
+        AZStd::wstring modelFile;
+
+        // If no filepath provided for onnx model, set default to a model.onnx file in the Assets folder.
+        if (initSettings.m_modelFile.empty())
+        {
+            // Get the FileIOBase to resolve the path to the ONNX gem
+            AZ::IO::FileIOBase* fileIo = AZ::IO::FileIOBase::GetInstance();
+            AZ::IO::FixedMaxPath onnxGemRoot;
+            fileIo->ResolvePath(onnxGemRoot, "@gemroot:ONNX@");
+            AZStd::string gemRootString = onnxGemRoot.c_str();
+            modelFile = AZStd::wstring(gemRootString.begin(), gemRootString.end()) + AZStd::wstring(L"/Assets/model.onnx");
+        }
+        else
+        {
+            modelFile = initSettings.m_modelFile;
+        }
+
         // If no model name is provided, will default to the name of the onnx model file.
         if (initSettings.m_modelName.empty())
         {
-            std::filesystem::path filepath = std::filesystem::path(initSettings.m_modelFile.c_str()).stem().c_str();
+            std::filesystem::path filepath = std::filesystem::path(modelFile.c_str()).stem().c_str();
             std::string filestring = filepath.string();
             AZ::StringFunc::Path::GetFileName(filestring.c_str(), m_modelName);
```

Comment:
This can use the equivlanet [AZ::IO::Path ](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L603-L607) class operations for retrieving the model name.

```
m_modelName = AZStd::to_wstring(onnxGemRoot.Stem());
```

Reply:
Changed